core/plugins/storage/ceph.py

```diff
@@ -514,6 +517,36 @@ def local_osds(self):
         self._local_osds = self._get_local_osds()
         return self._local_osds
 
+    def _get_pools(self, pools=None, crush_rule=-1):
+        res_pool = []
+        for pool in pools:
+            if pool['crush_rule'] == crush_rule:
+                pool_str = pool['pool_name'] + ' (' + str(pool['pool']) + ')'
+                res_pool.append(pool_str)
+
+        return res_pool
+
+    @property
+    def crush_rules(self):
+        """
+        Returns a list of crush rules, mapped to the respective pools.
+        """
```

Comment:
Since this is a property we should return self._crush_rules if it is already set so that we dont do the same work more than once e.g.

if self._crush_rules:
    return self._crush_rules

Reply:
done